library/src/androidTest/java/org/xmtp/android/library/ClientTest.kt

```diff
@@ -652,4 +654,81 @@ class ClientTest {
         assertEquals(client.inboxId, buildClient1.inboxId)
         assertEquals(client.inboxId, buildClient2.inboxId)
     }
+
+    @Test
+    fun testClientBackups() {
+        val fixtures = fixtures()
+        val key = SecureRandom().generateSeed(32)
+        val encryptionKey = SecureRandom().generateSeed(32)
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val alixWallet = PrivateKeyBuilder()
+
+        val alixClient = runBlocking {
+            Client().create(
+                account = alixWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    appContext = context,
+                    dbEncryptionKey = key
+                )
+            )
+        }
+
+        val directoryFile = File(context.filesDir.absolutePath, "testing_all")
+        val consentFile = File(context.filesDir.absolutePath, "testing_consent")
+
+        directoryFile.mkdir()
+        consentFile.mkdir()
+
+        val allPath = directoryFile.absolutePath + "/testAll.zstd"
+        val consentPath = consentFile.absolutePath + "/testConsent.zstd"
+
+        val boGroup = runBlocking {
+            val group = alixClient.conversations.newGroup(listOf(fixtures.boClient.address))
+            group.send("hi")
+            alixClient.conversations.syncAllConversations()
+            fixtures.boClient.conversations.syncAllConversations()
+            fixtures.boClient.findGroup(group.id)!!
+        }
+
+        val alixClient2 = runBlocking {
+            Client().create(
+                account = alixWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    appContext = context,
+                    dbEncryptionKey = key,
+                    dbDirectory = context.filesDir.absolutePath.toString()
+                )
+            )
+        }
+
+        runBlocking { alixClient.backupToFile(allPath, encryptionKey) }
+        runBlocking {
+            alixClient.backupToFile(
+                consentPath,
+                encryptionKey,
+                opts = BackupOptions(backupElements = listOf(BackupElement.CONSENT))
+            )
+        }
+
+        val metadataAll = runBlocking { alixClient.backupMetadata(allPath, encryptionKey) }
+        val metadataConsent = runBlocking { alixClient.backupMetadata(consentPath, encryptionKey) }
+
+        assertEquals(metadataAll.elements.size, 2)
+        assertEquals(metadataConsent.elements, listOf(BackupElement.CONSENT))
+
+        runBlocking {
+            alixClient2.importFromFile(allPath, encryptionKey)
+            alixClient2.conversations.syncAllConversations()
```

Comment:
@codabrink Another test failure here. It appears anytime I try to do anything with conversations after import it errors with this error

``` uniffi.xmtpv3.GenericException$GroupException: Group error: storage error: group with id e9dc3b6a01ea1e9eba787b3303e1136c not found ```